mantis-runtime-loader/src/main/java/io/mantisrx/runtime/loader/config/WorkerConfigurationWritable.java

```diff
@@ -80,6 +81,9 @@ public class WorkerConfigurationWritable implements WorkerConfiguration {
     @JsonIgnore
     MetricsCollector metricsCollector;
 
+    @JsonIgnore
+    ILeaderMonitorFactory leaderMonitorFactory;
```

Comment:
do you still need these in worker config?

Reply:
I think this does have to be here. The interface requires I add the field and I need some method to coerce to the `ILeaderMonitorFactory` interface, the test error is:

```
io.mantisrx.shaded.com.fasterxml.jackson.databind.exc.InvalidDefinitionException: Cannot construct instance of `io.mantisrx.server.core.ILeaderMonitorFactory` (no Creators, like default constructor, exist): abstract types either need to be mapped to concrete types, have custom deserializer, or contain additional type information
```

I think just keeping this ignore is within the current pattern of the workaround and least amount of work?

Reply:
The tricky part here is that the workconfig re-created from this class won't have the instances injected (as they are not created in the original frmProperties path).
So I think we got two options here:
1. Avoid instances and always rely on string to re-create instance via reflection.
2. manually set the instance for leaderMonitorFactory again at RuntimeTaskImpl (similar to the setMetricsCollector one.